consensus/types/src/beacon_state.rs

```diff
@@ -1627,6 +1627,22 @@ impl<T: EthSpec> BeaconState<T> {
         };
         Ok(sync_committee)
     }
+
+    /// Find the dependent roots for a given beacon state. These are the block roots that determine
+    /// shuffling. If one of these roots is updated, it means proposer and/or attester duties must be
+    /// recalculated.
+    ///
+    /// Returns a tuple `(dependent_root, previous_dependent_root)` where:
+    /// -  `dependent_root` is block root at the slot before the start slot of `state.current_epoch`.
+    /// -  `previous_dependent_root` is the block root at the slot before the start slot of `state.previous_epoch`.
+    pub fn get_dependent_roots(&self) -> Result<(Hash256, Hash256), Error> {
+        let dependent_root_slot = self.current_epoch().start_slot(T::slots_per_epoch()) - 1;
```

Comment:
Perhaps we could reduce some duplication with these functions:

https://github.com/sigp/lighthouse/blob/fff01b24ddedcd54486e374460855ca20d3dd232/consensus/types/src/beacon_state.rs#L574-L588

https://github.com/sigp/lighthouse/blob/fff01b24ddedcd54486e374460855ca20d3dd232/consensus/types/src/beacon_state.rs#L598-L616

Reply:
Should have known these existed :)